utils/writeable/src/lib.rs

```diff
@@ -159,9 +187,8 @@ macro_rules! assert_writeable_eq {
             use $crate::Writeable;
             let writeable = $actual_writeable;
             assert_eq!($expected_str, writeable.writeable_to_string());
-            if let $crate::LengthHint::Exact(len) = writeable.write_len() {
-                assert_eq!($expected_str.len(), len);
-            }
+            assert!(writeable.write_len().0 <= $expected_str.len());
+            writeable.write_len().1.map(|upper| assert!($expected_str.len() <= upper));
```

Comment:
Nit (optional): Here and below:

```suggestion
            if let Some(upper) = writeable.write_len().1 {
                assert!($expected_str.len() <= upper);
            }
```